depends/patches/qt/support_new_android_ndks.patch

```diff
@@ -0,0 +1,123 @@
+Follow Google's BuildSystemMaintainers doc to support future NDK releases.
+
+Upstream commit:
+ - Qt 5.14: 9b14950ff600a4ce5a8698b67ab38907c50417f1
+
+--- old/qtbase/mkspecs/android-clang/qmake.conf
++++ new/qtbase/mkspecs/android-clang/qmake.conf
+@@ -14,43 +14,33 @@
+ QMAKE_CC      = $$NDK_LLVM_PATH/bin/clang
+ QMAKE_CXX     = $$NDK_LLVM_PATH/bin/clang++
+ 
++# Follow https://android.googlesource.com/platform/ndk/+/ndk-release-r20/docs/BuildSystemMaintainers.md
++
+ equals(ANDROID_TARGET_ARCH, armeabi-v7a): \
+-    QMAKE_CFLAGS += -target armv7-none-linux-androideabi
+-else: equals(ANDROID_TARGET_ARCH, armeabi): \
+-    QMAKE_CFLAGS += -target armv5te-none-linux-androideabi
++    QMAKE_CFLAGS = -target armv7a-linux-androideabi$$replace(ANDROID_PLATFORM, "android-", "")
+ else: equals(ANDROID_TARGET_ARCH, arm64-v8a): \
+-    QMAKE_CFLAGS += -target aarch64-none-linux-android
++    QMAKE_CFLAGS = -target aarch64-linux-android$$replace(ANDROID_PLATFORM, "android-", "")
+ else: equals(ANDROID_TARGET_ARCH, x86): \
+-    QMAKE_CFLAGS += -target i686-none-linux-android -mstackrealign
++    QMAKE_CFLAGS = -target i686-linux-android$$replace(ANDROID_PLATFORM, "android-", "") -mstackrealign
+ else: equals(ANDROID_TARGET_ARCH, x86_64): \
+-    QMAKE_CFLAGS += -target x86_64-none-linux-android
++    QMAKE_CFLAGS = -target x86_64-linux-android$$replace(ANDROID_PLATFORM, "android-", "")
+ else: equals(ANDROID_TARGET_ARCH, mips): \
+-    QMAKE_CFLAGS += -target mipsel-none-linux-android
++    QMAKE_CFLAGS += -target mipsel-linux-android$$replace(ANDROID_PLATFORM, "android-", "")
```

Comment:
`+=` operator here unlike `=` in other `TARGET_ARCHES` above? 

Personally I'd get rid of `mips` as it's no longer suppored and to make the patchfile that much smaller.

Reply:
mips acrhs are removed in Qt 5.13 (https://github.com/qt/qtbase/commit/854156dd07ca5d50db9be091d34f68b9ca02ff21), but that commit is orthogonal to the issues solved here.

Reply:
> `+=` operator here unlike `=` in other `TARGET_ARCHES` above?

Thanks! [Updated](https://github.com/bitcoin/bitcoin/pull/22469#issuecomment-881861647).

Reply:
> Personally I'd get rid of mips as it's no longer suppored and to make the patchfile that much smaller.

Inclined to agree with @icota here. In any case, if mips support was removed in 5.13, and the patch here is from 5.14, why does it contain anything mips related? If you've added that to the patch yourself, why isn't that mentioned / documented anywhere?

Reply:
Agree.

> Personally I'd get rid of `mips` as it's no longer suppored and to make the patchfile that much smaller.

> Inclined to agree with @icota here.

[Updated](https://github.com/bitcoin/bitcoin/pull/22469#issuecomment-882055056).